### DISCUSSION TITLE
Feature Request: Customizable delay before initial browser scan

### DISCUSSION THREAD

**Discussion by i0ntempest** [will-stone/browserosaurus#489] [Ideas]
Since the last update B only scans for browsers when clicking on a link for the 1st time, which is a bit annoying since it takes a while. I tried to get used to it but I really can't. I'm aware the change is because sometimes if the scan is run before spotlight index is done the result wouldn't be complete, therefore I would suggest adding a customizable delay in settings before the first scan is run.
Or if for some reason that can't be done, a fixed 5-minute delay should allow indexing on whatever machine to complete I guess?

**Comment by will-stone** [comment]
Hi @i0ntempest is there any reason why you aren't leaving Browserosaurus (and your Mac) running? Older Mac or something like that?

**Comment by i0ntempest** [comment]
I do keep it running most of the time but I often switch between OSes, both on my Hack and my Mac mini.

**Reply by will-stone** [reply]
Oh, that’s quite a niche use case. As you’ve probably read from my comments on other feature requests: I’m very loath to add configuration, especially configuration for something that doesn’t affect me. This may sound selfish but any open source dev will tell you that maintaining code that you do not have use for is not fun. I have an intel work MacBook which, I agree, is slow to load the browsers on first load, but I only reboot once in a blue moon. My personal machine is a MacBook Air M1 and it loads the browsers in about a second 😮 I don’t know why it’s soooo much faster on an M1 but I’m guessing it may be an electron bug. Therefore, let’s wait and see if future updates fix this delay. Having said all that, I could show last known configuration whilst it’s scanning 🤔 Let me have a think about it.

**Reply by i0ntempest** [reply]
> Oh, that’s quite a niche use case. As you’ve probably read from my comments on other feature requests: I’m very loath to add configuration, especially configuration for something that doesn’t affect me. This may sound selfish but any open source dev will tell you that maintaining code that you do not have use for is not fun. I have an intel work MacBook which, I agree, is slow to load the browsers on first load, but I only reboot once in a blue moon. My personal machine is a MacBook Air M1 and it loads the browsers in about a second 😮 I don’t know why it’s soooo much faster on an M1 but I’m guessing it may be an electron bug. Therefore, let’s wait and see if future updates fix this delay. Having said all that, I could show last known configuration whilst it’s scanning 🤔 Let me have a think about it.

I wouldn't say dual booting is a niche use case... but I don't complain, B is free after all. I would gladly accept showing last known as a solution.

**Reply by will-stone** [reply]
Well, obviously I can't condone Hackintoshes 😜 Which leaves dual booting Windows and Mac, which I can see as something you'd want to switch between fairly regularly, but I would have thought the majority of people stick to macOS, especially now ARM macs don't support dual booting (to my knowledge). Anyway...

Give 17.1.0 a go. It'll save your apps' installed status to the store, and then show those apps whilst scanning in the background. I've not tried it on my intel machine yet so please file an issue if you encounter any. And please note you'll need to have run 17.1.0 _before_ this functionality will work, so: upgrade -> relaunch -> quit -> start -> last known installed apps should show from now on 🥳

**Reply by i0ntempest** [reply]
Awesome, just tried and no problems so far. Thank you.

**Reply by will-stone** [reply]
No worries. Between this and the Spotlight delay issue, we now have an app that is responsive immediately, so thank you for pushing me to think further 😎
